driver-core/src/main/java/com/datastax/driver/core/DefaultResultSetFuture.java

```diff
@@ -156,7 +156,7 @@ public void run() {
                     setException(new DriverInternalError(String.format("Got unexpected %s response from %s", response.type, connection.address)));
                     break;
             }
-        } catch (RuntimeException e) {
+        } catch (Throwable e) {
```

Comment:
Not a big fan of this, are you trying to catch the `AssertionError` thrown by `ArrayBackedResultSet`?